astropy/io/fits/fitsrec.py

```diff
@@ -478,9 +478,22 @@ def from_columns(cls, columns, nrows=0, fill=False, character_as_bytes=False):
         return data
 
     def __repr__(self):
-        # Force use of the normal ndarray repr (rather than the new
-        # one added for recarray in Numpy 1.10) for backwards compat
-        return np.ndarray.__repr__(self)
+        repr_dtype = self.dtype
+        # FIXME: remove numpy.record in the dtype repr, as recarray?
```

Comment:
If you're editing anyway, maybe move this together with the comment I suggested adding above.